src/intl-head.tsx

```diff
@@ -0,0 +1,28 @@
+import Head from 'next/head';
+import { useRouter } from 'next/router';
+import React, { ReactElement, ReactNode } from 'react';
+import { useAlternateLinks } from './hooks/useAlternateLinks';
+import { useCanonicalUrl } from './hooks/useCanonicalUrl';
+
+interface IntlHeadProps {
+  title: string;
+  children?: ReactNode;
+  currentLocale?: string;
+}
+
+export function IntlHead({ children, title, currentLocale }: IntlHeadProps): ReactElement {
+  const { locale } = useRouter();
+  const canonicalUrl = useCanonicalUrl(currentLocale ?? locale);
+  const alternateLinks = useAlternateLinks(currentLocale ?? locale);
+
+  return (
+    <Head>
+      <title>{title}</title>
+      {alternateLinks.map(({ href, hrefLang }) => (
+        <link rel="alternate" href={`${href}`} hrefLang={hrefLang} key={hrefLang} />
+      ))}
+      <link rel="canonical" href={canonicalUrl} />
```

Comment:
I'm not sure that this canonical URL link is strictly within the scope of the npm module. It's important for SEO, sure, but it's got nothing to do with localisation or internationalisation.

Reply:
I understand but these are the requirements provided me by @nbouvrette. I don't know if we should move it to the example instead. I'll let him decide.

Reply:
So after discussions with Nicolas, it was decided that it should be removed from IntlHead, I'm going to move `useCanonicalUrl` to the example.